### ISSUE TITLE
licensing review: drop non-commercial and no-derivates terms?

### ISSUE THREAD

**Issue by dkg** [py-pdf/sample-files#18]
Over in https://bugs.debian.org/1028570, i observed that the licensing for these files is not compatible with the debian free software guidelines (DFSG).

Is it possible to relicense it from CC 4.0 BY-NC-ND to just CC 4.0 BY?  or CC 4.0 BY-SA?

I contributed `015-arabic/*` and I'm certainly fine with those files being relicensed in such a way.  I'm not sure of the origins of the other files in question.

Without relicensing, it's going to be a challenge to ensure that the package's tests run correctly in Debian, since all build and test infrastructure in the project is intended to use only DFSG-free material.

**Comment by MartinThoma** [comment]
The van run the tests without the sample files.

I have to think about changing the license. It might be that only the two of us have contributed so far.

**Comment by dkg** [comment]
Not sure what you mean by "the van run the tests" -- can you elaborate?  Are you saying i should just run the tests without the sample files?  I assume there will be less coverage in that case.

If it's possible to change the license, that would simplify matters a lot for me.  Thanks for considering it!

**Comment by dkg** [comment]
fwiw, the scope of coverage is already limited because debian test suites don't use the internet at all , and we have [a patch to disable the network-reliant tests](https://salsa.debian.org/debian/pypdf/-/blob/debian/pypdf2/unstable/debian/patches/disable-network-tests.patch).  The more the tests can be done on local copies of files that share similar licensing terms to the software itself, the more likely we'll be able to have the full test suite run on all the different architectures that debian's testing infrastructure supports.

**Comment by MartinThoma** [comment]
I don't see the point of running the tests on different architectures as this is all about Python. If (and only if) it works for any architecture with a given Python version, it will always work for all architectures with that Python version.

**Comment by MartinThoma** [comment]
Or in other words: `pypdf` runs its tests on Python 3.6 - Python 3.11 (see [GitHub CI](https://github.com/py-pdf/pypdf/blob/main/.github/workflows/github-ci.yaml#L24)) and officially supports 3.6 - 3.11 (see [trove classifiers](https://github.com/py-pdf/pypdf/blob/main/pyproject.toml#L19-L24)).

The main thing that might be interesting to test is stuff that requires additional packages (pycryptodome / Pillow) to check if the Debian packaging broke anything :thinking:  I'm not sure how that actually works.

**Comment by MartinThoma** [comment]
I hope I find the time at the end of the day to go over all files / make up my mind if this change of license is possible / a good idea.

The license you propose would be more permissive, but people could sell the stuff in there without having ever contributed, right?

**Comment by dkg** [comment]
Yes, with standard CC BY licensing it would be completely permissive.  with CC BY-SA it would be closer to a copyleft situation: anyone transmitting the file to anyone else would need to "share alike".

**Comment by dkg** [comment]
Any progress on this?  In debian, i've moved ahead by just dropping tests on these sample files, but if they get a DFSG-free license, i'd be happy to ship them as a corpus and the nexpand the pypdf test suite; or we could include them as a "second tarball" for the source package for pypdf directly.

**Comment by MartinThoma** [comment]
Sorry, I forgot this one. Let's go through it:

* https://github.com/py-pdf/sample-files/blob/main/014-outlines/mistitled_outlines_example.pdf @mtd91429 Would it be ok for you if we changed the license of this repository from [`CC 4.0 BY-NC-ND`](https://creativecommons.org/licenses/by-nc-nd/4.0/deed.en) to [`CC 4.0 BY-SA`](https://creativecommons.org/licenses/by-sa/4.0/deed.en)?
* https://github.com/py-pdf/sample-files/tree/main/017-unreadable-meta-data - @ediamondscience Would it be ok for you if we changed the license of this repository from [`CC 4.0 BY-NC-ND`](https://creativecommons.org/licenses/by-nc-nd/4.0/deed.en) to [`CC 4.0 BY-SA`](https://creativecommons.org/licenses/by-sa/4.0/deed.en)?
* https://github.com/py-pdf/sample-files/tree/main/018-base64-image - @mergezalot Would it be ok for you if we changed the license of this repository from [`CC 4.0 BY-NC-ND`](https://creativecommons.org/licenses/by-nc-nd/4.0/deed.en) to [`CC 4.0 BY-SA`](https://creativecommons.org/licenses/by-sa/4.0/deed.en)?
* https://github.com/py-pdf/sample-files/tree/main/019-grayscale-image - @joeywang4 Would it be ok for you if we changed the license of this repository from [`CC 4.0 BY-NC-ND`](https://creativecommons.org/licenses/by-nc-nd/4.0/deed.en) to [`CC 4.0 BY-SA`](https://creativecommons.org/licenses/by-sa/4.0/deed.en)?

**Comment by mergezalot** [comment]
Hello @MartinThoma, 

yes. CC 4.0 BY-SA for https://github.com/py-pdf/sample-files/tree/main/018-base64-image is fine. 

regards

**Comment by joeywang4** [comment]
Hi @MartinThoma,

Sure, that's fine.

**Comment by ediamondscience** [comment]
@MartinThoma It's fine with me too.

**Comment by dkg** [comment]
@mtd91429 can you follow up here?

**Comment by stefan6419846** [comment]
@mtd91429 Could you please review whether you are fine with the license for the file (modification) you did?

**Comment by mtd91429** [comment]
Sorry for the delayed response - yes it is fine by me to alter the license.
